qiskit/providers/backend_compat.py

```diff
@@ -13,24 +13,367 @@
 """Backend abstract interface for providers."""
 
 from __future__ import annotations
-
-from typing import List, Iterable, Any, Dict, Optional
+import logging
+from typing import List, Iterable, Any, Dict, Optional, Union, Tuple
 
 from qiskit.exceptions import QiskitError
 
 from qiskit.providers.backend import BackendV1, BackendV2
 from qiskit.providers.backend import QubitProperties
 from qiskit.utils.units import apply_prefix
-from qiskit.circuit.library.standard_gates import get_standard_gate_name_mapping
-from qiskit.circuit.measure import Measure
+from qiskit.providers.models.backendconfiguration import (
+    QasmBackendConfiguration,
+    PulseBackendConfiguration,
+)
+
 from qiskit.providers.models.backendconfiguration import BackendConfiguration
 from qiskit.providers.models.backendproperties import BackendProperties
+
 from qiskit.providers.models.pulsedefaults import PulseDefaults
 from qiskit.providers.options import Options
 from qiskit.providers.exceptions import BackendPropertyError
+from qiskit.providers.models.backendproperties import Gate as GateSchema
+
+
+logger = logging.getLogger(__name__)
+
+
+class IBMQubitProperties(QubitProperties):
+    """A representation of the properties of a qubit on an IBM backend."""
+
+    __slots__ = (  # pylint: disable=redefined-slots-in-subclass
+        "t1",
+        "t2",
+        "frequency",
+        "anharmonicity",
+        "operational",
+    )
+
+    def __init__(  # type: ignore[no-untyped-def]
+        self,
+        t1=None,
+        t2=None,
+        frequency=None,
+        anharmonicity=None,
+        operational=True,
+    ):
+        """Create a new ``IBMQubitProperties`` object
+
+        Args:
+            t1: The T1 time for a qubit in secs
+            t2: The T2 time for a qubit in secs
+            frequency: The frequency of a qubit in Hz
+            anharmonicity: The anharmonicity of a qubit in Hz
+            operational: A boolean value representing if this qubit is operational.
+        """
+        super().__init__(t1=t1, t2=t2, frequency=frequency)
+        self.anharmonicity = anharmonicity
+        self.operational = operational
+
+    def __repr__(self):  # type: ignore[no-untyped-def]
+        return (
+            f"IBMQubitProperties(t1={self.t1}, t2={self.t2}, frequency={self.frequency}, "
+            f"anharmonicity={self.anharmonicity})"
+        )
 
 
 def convert_to_target(
+    configuration: Union[QasmBackendConfiguration, PulseBackendConfiguration],
+    pulse_defaults: Optional[Dict] = None,
+    properties: Optional[Dict] = None,
```

Comment:
```suggestion
    configuration: BackendConfiguration,
    properties: Optional[Union[BackendProperties,Dict]] = None,
    defaults: Optional[Union[PulseDefaults,Dict]] = None,
    custom_name_mapping: Optional[Dict[str, Any]] = None,
    add_delay: bool = False,
    filter_faulty: bool = False,
```

Sudden API changes would cause errors and it would be better to set both the type you want to drop and you want to add in the next release.
Regarding `properties` and `defaults`, if inputs are not `Dict`, you can add codes that change `properties` and `defaults` to `Dict`.(you can use https://github.com/Qiskit/qiskit/blob/02cb814e7669b8035924399b935b003032e1f7b3/qiskit/providers/models/backendproperties.py#L253.)

Regarding `custom_name_mapping`, `add_delay`, and `filter_faulty`, if the new convert_to_target doesn't have the same functionality as the legacy one, could you please add the functionality to this function?

Reply:
I have added the same functionality.
All arguments of `convert_to_target` are relevant and now works fine to my knowledge :)

Reply:
As I tried to use `convert_to_target` for `BackendV2Converter` I realized your this suggestion holds a lot of weight, all arguments are necessary.
But here again makes it a bit dependent on `BackendConfiguration`, `BackendProperties` and `PulseDefaults`